src/Sticky.jsx

```diff
@@ -276,6 +276,12 @@ class Sticky extends Component {
         this.update();
     }
 
+    componentDidUpdate(prevProps, prevState) {
+        if (prevState.status !== this.state.status) {
+            this.props.onStickyStateChange(this.state.status === 2);
```

Comment:
Fixed the first two issues.

As for returning the status, I originally did it like that, but then thought that it's leaking an implementation detail. More abstract to just return whether or not it's currently sticky.
I can change it if you would like. The idea of exporting the statuses as constants sounds good.